lnwallet/channel.go

```diff
@@ -1790,6 +1790,30 @@ func (lc *LightningChannel) restoreStateLogs(
 		outgoingLocalAddHeights[l.HtlcIndex] = localCommitment.height
 	}
 
+	// If we have any unsigned acked updates to sign for, then the add is no
+	// longer on our local commitment, but is still on the remote's commitment.
+	// <---fail---
+	// <---sig----
+	// ----rev--->
+	// To ensure proper channel operation, we restore the add's addCommitHeightLocal
+	// field to the height of our local commitment.
+	for _, logUpdate := range unsignedAckedUpdates {
+
+		var htlcIdx uint64
+		switch wireMsg := logUpdate.UpdateMsg.(type) {
+		case *lnwire.UpdateFulfillHTLC:
+			htlcIdx = wireMsg.ID
+		case *lnwire.UpdateFailHTLC:
+			htlcIdx = wireMsg.ID
+		case *lnwire.UpdateFailMalformedHTLC:
+			htlcIdx = wireMsg.ID
+		default:
+			continue
+		}
+
+		outgoingLocalAddHeights[htlcIdx] = localCommitment.height
```

Comment:
add a comment that this value will be used to set the local add height on the outgoing HTLCs on the remote commitment below.

Reply:
comment added